### PR TITLE
IGAPP-428: Inner categories with timestamp

### PR DESCRIPTION
**Pull request by Cmd190**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by Cmd190** [comment]
Ihr könnt mir gerne noch weitere Reviewer empfehlen

**Comment by Cmd190** [comment]
Eine Frage hätte ich noch: 

Könnte man in der CategoryList

```
 {content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }}
                                   onInternalLinkClick={onInternalLinkClick} />}
        {content && lastUpdate &&
         <LastUpdateInfo lastUpdate={lastUpdate} formatter={formatter} withText />}
```

zu 

`{content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }}
                                   onInternalLinkClick={onInternalLinkClick} />
         && lastUpdate &&
         <LastUpdateInfo lastUpdate={lastUpdate} formatter={formatter} withText />}` 
vereinfachen oder würde das das Rendern des Content verhindern, wenn lastUpdate == false ?

**Comment by Cmd190** [comment]
Bei mir lokal laufen die Tests durch

**Comment by klinzo** [comment]
> Bei mir lokal laufen die Tests durch

You have to check linting and flow types as well. You can check this by:
`yarn lint --quiet` and
`yarn flow`. If flow errors do not go away even though you are sure it should, you may have to execute `yarn flow:check-now`.

There should be support for highlighting these errors in vscode as well. Did you follow the steps in our vscode docs?https://github.com/Integreat/integreat-app/blob/main/docs/vscode.md

**Comment by klinzo** [comment]
> Eine Frage hätte ich noch:
> 
> Könnte man in der CategoryList
> 
> ```
>  {content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }}
>                                    onInternalLinkClick={onInternalLinkClick} />}
>         {content && lastUpdate &&
>          <LastUpdateInfo lastUpdate={lastUpdate} formatter={formatter} withText />}
> ```
> 
> zu
> 
> `{content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }} onInternalLinkClick={onInternalLinkClick} /> && lastUpdate && <LastUpdateInfo lastUpdate={lastUpdate} formatter={formatter} withText />}`
> vereinfachen oder würde das das Rendern des Content verhindern, wenn lastUpdate == false ?

I think that would not work as only the last expression is really rendered afaik. But feel free to try it for yourself.
Imo the way it is currently is also simpler and easier to understand.

**Review by klinzo — COMMENTED** [review]
Looking good apart from the flow types :)

**Review by Taggotty — COMMENTED** [review]
👍🏼 Good work so far. Take another look at SearchPage.js. There we also use the category list.
You can also consider passing the whole category to the CategoryList, I think. Otherwise there will probably be a lot of optional parameters where it is not necessary. If you struggle with the formatter in the SearchPage.js take a look at Page.js :)

**Comment by Cmd190** [comment]
> > Eine Frage hätte ich noch:
> > Könnte man in der CategoryList
> > ```
> >  {content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }}
> >                                    onInternalLinkClick={onInternalLinkClick} />}
> >         {content && lastUpdate &&
> >          <LastUpdateInfo lastUpdate={lastUpdate} formatter={formatter} withText />}
> > ```
> > 
> > 
> > zu
> > `{content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }} onInternalLinkClick={onInternalLinkClick} /> && lastUpdate && <LastUpdateInfo lastUpdate={lastUpdate} formatter={formatter} withText />}`
> > vereinfachen oder würde das das Rendern des Content verhindern, wenn lastUpdate == false ?
> 
> I think that would not work as only the last expression is really rendered afaik. But feel free to try it for yourself.
> Imo the way it is currently is also simpler and easier to understand.

I have all required extensions, but flow is not working

**Comment by Cmd190** [comment]
> > Bei mir lokal laufen die Tests durch
> 
> You have to check linting and flow types as well. You can check this by:
> `yarn lint --quiet` and
> `yarn flow`. If flow errors do not go away even though you are sure it should, you may have to execute `yarn flow:check-now`.
> 
> There should be support for highlighting these errors in vscode as well. Did you follow the steps in our vscode docs?https://github.com/Integreat/integreat-app/blob/main/docs/vscode.md

All required extensions are installed, but flow is not working

**Comment by Cmd190** [comment]
> 👍🏼 Good work so far. Take another look at SearchPage.js. There we also use the category list.
> You can also consider passing the whole category to the CategoryList, I think. Otherwise there will probably be a lot of optional parameters where it is not necessary. If you struggle with the formatter in the SearchPage.js take a look at Page.js :)

I looked at SearchPage.js but my changes do not seem to have any influence on the usage of CategoryList there. I think there is no need to modify SearchPage.js ?

**Review by maxammann — APPROVED** [review]

**Comment by maxammann** [comment]
Nice and clean PR! Make sure to prefix your commits: `IGAPP-428: asdf`

File: web/src/routes/categories/components/CategoryList.js [hunk 1]
```diff
@@ -42,13 +42,17 @@ type PropsType = {|
...
-    const { categories, title, thumbnail, content, query, onInternalLinkClick } = this.props
+    const {
+      categories, query, onInternalLinkClick, formatter, category
+    } = this.props
     return (
       <div>
-        {thumbnail && <Centering><CategoryIcon src={thumbnail} alt='' /></Centering>}
-        {title && <Caption title={title} />}
-        {content && <RemoteContent dangerouslySetInnerHTML={{ __html: content }}
+        {category && category.thumbnail && <Centering><CategoryIcon src={category.thumbnail} alt='' /></Centering>}
```
Comment (klinzo) [inline web/src/routes/categories/components/CategoryList.js:50]: Alternatively you could also do which is cleaner imo, same for other checks:
```suggestion
        {category?.thumbnail && <Centering><CategoryIcon src={category.thumbnail} alt='' /></Centering>}
```
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining for more infos.

**Review by klinzo — APPROVED** [review]

**Pull request merged**